src/core/send.c

```diff
@@ -1498,6 +1498,13 @@ QuicSendFlush(
         //QuicConnUpdatePeerPacketTolerance(Connection, Builder.TotalCountDatagrams);
     }
 
+    //
+    // Cancel all outstanding send requests if not able to send.
+    //
+    if(!(Send->SendFlags & QUIC_SEND_FLAG_DGRAM_CANCEL_ON_BLOCKED)) {
```

Comment:
These are different kinds of send flags, so you need to just always call this function (no `if` check).